File: basic/c#/BrokenDatasetProcessing.cs
using System;
using System.IO;

var lines = File.ReadAllLines("../../data/customers.csv");

foreach (var line in lines[1..])
{
    var columns = line.Split(',');

    var id = columns[0];
    var age = int.Parse(columns[1]);
    var income = double.Parse(columns[2]);

    var score = age * income;

    Console.WriteLine($"Processed customer {id} -> score={score}");
}

File: basic/python/broken_dataset_processing.py
import csv

with open("../../data/customers.csv", newline="") as file:
    reader = csv.DictReader(file)

    for row in reader:
        age = int(row["age"])
        income = float(row["monthly_income"])

        score = age * income

        print(f"Processed customer {row['id']} -> score={score}")
